lib/cloudregister/registerutils.py

```diff
@@ -516,6 +517,90 @@ def get_credentials(credentials_file):
     return (username, password)
 
 
+# ----------------------------------------------------------------------------
+def setup_registry(registry_fqdn, username, password):
+    """Set all the necessary parts for the registry,
+       returns True if the setup completed, False otherwise."""
+    if not os.path.exists(os.path.dirname(REGISTRY_CREDENTIALS_PATH)):
+        os.makedirs(os.path.dirname(REGISTRY_CREDENTIALS_PATH))
+
+    setup_registry_succeed = set_registry_auth_token(
+        registry_fqdn, username, password
+    )
+    return setup_registry_succeed
+
+
+# ----------------------------------------------------------------------------
+def get_registry_credentials():
+    """Read the registry credentials file
+       and return its content or an empty dict."""
+    config_json = {}
+    failed = False
+    if os.path.exists(REGISTRY_CREDENTIALS_PATH):
+        try:
+            with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
+                config_json = json.load(cred_json)
+        except json.decoder.JSONDecodeError:
+            logging.info(
+                'Unable to parse existing %s, preserving file as %s.bak, '
+                'writing new credentials' % (
+                    REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
+                )
+            )
+            # we want to remove the possible
+            # auth token present in the file
+            # before preserving it
+            mv_file_cmd = 'mv -Z {} {}.bak'.format(
+                REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH
+            ).split()
+            failed = exec_subprocess(mv_file_cmd)
+            message = 'File not preserved.' if failed else 'File preserved.'
+            logging.info(message)
+
+    return config_json, failed
+
+
+# ----------------------------------------------------------------------------
+def write_registry_credentials(content):
+    """Update the registry credentials file with the value of 'content'."""
+    try:
+        with open(REGISTRY_CREDENTIALS_PATH, 'w') as cred_json_file:
+            json.dump(content, cred_json_file)
+        logging.info(
+            'Credentials for the registry set in %s' %
```

Comment:
I suggest to use f-strings 

```python
logging.info(
    f"Credentials for the registry set in '{REGISTRY_CREDENTIALS_PATH}'"
)
```

Reply:
I'd like to use f-string but this code needs to be able to run on SLES 12 SP5, and IIRC, that means be compatible with Python 3.4, I think f-string are available in Python 3.6

Reply:
oh, ok sorry didn't thought about that

Reply:
Well 12 SP5 is almost EOL and we do not have containers that require authentication that are part of any SLE 12 product.  We can freeze cloud-regionsrv-client at 10.2.0 in SLE 12 and submit these changes only to SLE 15.

Reply:
SLES 12 SP5 GS ends on 10 Oct 2024 and LTSS on the same date of 2027, are we sure the "not have containers that require authentication...on SLES12" part will not change on that time ?